devicemodel/hw/pci/passthrough.c

```diff
@@ -1073,21 +1072,14 @@ passthru_write(struct vmctx *ctx, int vcpu, struct pci_vdev *dev, int baridx,
 	       uint64_t offset, int size, uint64_t value)
 {
 	struct passthru_dev *ptdev;
-	struct iodev_pio_req pio;
 
 	ptdev = dev->arg;
 
 	if (baridx == ptdev_msix_table_bar(ptdev)) {
 		msix_table_write(ptdev, offset, size, value);
 	} else {
-		assert(dev->bar[baridx].type == PCIBAR_IO);
-		bzero(&pio, sizeof(struct iodev_pio_req));
-		pio.access = IODEV_PIO_WRITE;
-		pio.port = ptdev->bar[baridx].addr + offset;
-		pio.width = size;
-		pio.val = value;
-
-		(void)ioctl(iofd, IODEV_PIO, &pio);
+		/* TODO: Add support for IO BAR of PTDev */
+		warnx("Passthru: PIO write not supported, ignored\n");
```

Comment:
So this may happen, right? Can we gracefully handle it?
Also, the patch changed the semantics, make sure this is what we want.

Reply:
Actually, PIO BAR access is not emulated, the ioctl is not implemented in sos kernel. For now, there is no device accesses PIO BAR.
But it can not be guaranteed on different platforms. Eventually, we probably need to add the emulation for PIO BAR. But for this patch, it should be fine.